sdk/inc/azure/core/az_result.h

```diff
@@ -80,6 +80,9 @@ typedef enum
   /// Not supported.
   AZ_ERROR_NOT_SUPPORTED = _az_RESULT_MAKE_ERROR(_az_FACILITY_CORE, 7),
 
+  /// An external dependency required to perform operation was is not provided.
+  AZ_ERROR_DEPENDENCY_NOT_PROVIDED = _az_RESULT_MAKE_ERROR(_az_FACILITY_CORE, 8),
```

Comment:
Should this go under the platform facility?

Reply:
Also, consider adding a change log entry for this addition/change.

Reply:
You could argue either way, but the error is general in nature. The same way, other core codes, if you look at them are general in nature, but other modules may return them. I'd say the module is not about the point where the error was originated, but about its being specific to a particular scope.

Reply:
Makes sense. This one is sort of like the out-of-memory one though (which is under platform). Since core doesn't have any dependencies, it is odd to have one with that facility.

The facility is internal, if you feel strongly, we can defer changing it to later.